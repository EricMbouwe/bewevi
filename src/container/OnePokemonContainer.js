import React from 'react';
import { useSelector } from 'react-redux';
import OnePokemon from '../component/OnePokemon';
import emptyImageSVG from '../assets/emptyImage.svg';
import Loading from '../component/Loading';

const OnePokemonContainer = () => {
  const pokemon = useSelector(state => state.pokemon.chosen);
  const loading = useSelector(state => state.pokemon.loading);

  const handleError = e => {
    e.target.src = emptyImageSVG;
  };
  console.warn(pokemon);
  return (
    <>
      {loading && <Loading />}
      {pokemon.id ? (
        <OnePokemon
          pokemon={pokemon}
          handleError={handleError}
        />
      ) : (
        <div />
      )}
    </>
  );
};

export default OnePokemonContainer;
