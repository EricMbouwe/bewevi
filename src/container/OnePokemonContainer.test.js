import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { useSelector } from 'react-redux';

import ReactTestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import OnePokemonContainer from './OnePokemonContainer';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock('../component/OnePokemon');

beforeEach(() => {
  useSelector.mockImplementation(selector => selector(
    {
      pokemon: {
        chosen: {
          common_name: 'Ivy common',
          scientific_name: 'Ivy scientific',
          image_url: 'Ivy.jpg',
          images: {
            fruit: [Error()],
            leaf: ['image.jpg'],
            flower: ['image.jpg'],
          },
        },
        loading: false,
      },
    },
  ));
});

describe('<PokemonsListContainer />', () => {
  it('is connecting to store', () => {
    render(
      <OnePokemonContainer />,
    );
    expect(screen.getByText(/Ivy scientific/i)).toBeInTheDocument();
    expect(screen.getByText(/Ivy common/i)).toBeInTheDocument();
  });

  it('is updating if there is an error when images are being installed', () => {
    render(
      <OnePokemonContainer />,
    );
    expect(screen.getByAltText('test').src).toEqual('http://localhost/#');

    ReactTestUtils.Simulate.error(screen.getByAltText('test'));
    expect(screen.getByAltText('test').src).toEqual('http://localhost/emptyImage.svg');
  });

  it('renders Loading if loading is \'false\' in store', () => {
    useSelector.mockImplementation(selector => selector(
      {
        pokemon:
        {
          chosen: { },
          loading: true,
        },
      },
    ));
    const tree = renderer.create(<OnePokemonContainer />).toJSON();
    expect(tree[0].props.className).toEqual('loading main');
  });

  it('renders correctly', () => {
    const renderedContainer = render(
      <OnePokemonContainer />,
    );
    expect(renderedContainer).toMatchSnapshot();
  });
});
