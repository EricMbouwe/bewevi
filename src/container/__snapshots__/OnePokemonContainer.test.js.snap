// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonsListContainer /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      {"common_name":"Ivy common","scientific_name":"Ivy scientific","image_url":"Ivy.jpg","images":{"fruit":[{}],"leaf":["image.jpg"],"flower":["image.jpg"]}}
      <img
        alt="test"
        src="#"
      />
    </div>
  </body>,
  "container": <div>
    {"common_name":"Ivy common","scientific_name":"Ivy scientific","image_url":"Ivy.jpg","images":{"fruit":[{}],"leaf":["image.jpg"],"flower":["image.jpg"]}}
    <img
      alt="test"
      src="#"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
